popup.js

```diff
@@ -0,0 +1,157 @@
+const submitBtn = document.getElementById('submit-btn')
+const addProjBtn = document.getElementById('addproject-btn')
+const urlBtn = document.getElementById('url-btn')
+const addProjKeyInput = document.getElementById('addProjectKey-input')
+const addProjKeyBtn = document.getElementById('addProjectKey-btn')
+const cancelBtn = document.getElementById('cancel-btn')
+const keysModal = document.getElementById('keys-modal')
+const urlModal = document.getElementById('url-modal')
+const keyList = document.getElementById('keyList')
+const modalBackground = document.getElementById('modal-background')
+const issueIdInput = document.getElementById('id-input')
+const urlComposition = document.getElementById('url-composition')
+const projectKeySelect = document.getElementById('project-select')
+const saveJiraUrlBtn = document.getElementById('save-url-btn')
+const deleteBtn = document.getElementById('delete-btn')
+const companyInput = document.getElementById('company-input')
+const jiraBaseUrl = document.getElementById('jira-url-input')
+//let jiraUrl = ""
+let storedKeys = ["1"]
+
+addProjBtn.addEventListener('click', () => controlModalDisplayStyle("block", keysModal))
+urlBtn.addEventListener('click', () => controlModalDisplayStyle("block", urlModal))
+saveJiraUrlBtn.addEventListener('click', () => {
+    storeJiraUrlElements(companyInput.value, jiraBaseUrl.value)
+    controlModalDisplayStyle("none", urlModal)
+})
+cancelBtn.addEventListener('click', () => {
+    controlModalDisplayStyle("none", keysModal)
+    resetInputValue(addProjKeyInput);
+})
+addProjKeyBtn.onclick = addProjectKey;
+submitBtn.onclick = () => openNewTab(issueIdInput.value);
+window.onload = () => {
+    useStoredOptionsForDisplayInDOM()
+    populateUrlFieldsWithUrlElements()
+}
+projectKeySelect.onchange = makeDropDownSelectionDefault;
+issueIdInput.onkeyup = () => {
+    if (event.keyCode === 13) {
+        openNewTab(issueIdInput.value);
+     }
+}
+companyInput.addEventListener('keyup',function(){
+    updateJiraUrlInUrlCompositionDom()
+});
+jiraBaseUrl.addEventListener('keyup',function(){updateJiraUrlInUrlCompositionDom()});
+
+
+function openNewTab(inputValue) {
+    let url = generateIssueUrl(inputValue)
+    chrome.tabs.create({'url': url});
+}
+
+// Puts latest selection to top to make it default one for page reloads
+function makeDropDownSelectionDefault(){
+    // TODO: make general function that takes param and makes it first array element
+    storedKeys = storedKeys.filter(item => item !== projectKeySelect.value);
+    storedKeys.unshift(projectKeySelect.value);
+    storeProjectKeys(storedKeys)
+}
+
+function generateIssueUrl(inputValue) {    
+    return `https://${companyInput.value}.${jiraBaseUrl.value + projectKeySelect.value}${inputValue ? '-' + issueIdInput.value : ''}`
```

Comment:
If the JIRA id input field is empty no dash is added so only the project page opens. This is declared nowhere. Should it and if so how? In a comment or better variable naming?